odoo/orm/query.py

```diff
@@ -140,6 +141,15 @@ def left_join(self, lhs_alias: str, lhs_column: str, rhs_table: str, rhs_column:
         self.add_join('LEFT JOIN', rhs_alias, rhs_table, condition)
         return rhs_alias
 
+    def add_temp_table(self, alias: str, subquery: SQL):
```

Comment:
The name is really bad: this adds a CTE not a temporary table.

You also have no support for RECURSIVE and NOT MATERIALIZED.
https://www.postgresql.org/docs/16/queries-with.html#QUERIES-WITH-CTE-MATERIALIZATION